common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryOAuth2Strategy.java

```diff
@@ -54,10 +54,15 @@
         TokenResult> {
 
     private AzureActiveDirectoryOAuth2Configuration mConfig = null;
+    private static final int HTTP_BAD_REQUEST = 400;
 
+    /**
+     * Constructor of AzureActiveDirectoryOAuth2Strategy.
+     * @param config Azure Active Directory OAuth2 configuration
+     */
     public AzureActiveDirectoryOAuth2Strategy(final AzureActiveDirectoryOAuth2Configuration config) {
         super(config);
-        mTokenEndpoint = "https://login.microsoftonline.com/microsoft.com/oauth2/token";
+        this.setTokenEndpoint("https://login.microsoftonline.com/microsoft.com/oauth2/token");
```

Comment:
constructors (of non-final classes) should call only final or private methods. 

Categorically, I really dislike it when non-`final` classes has constructors which call non-`final` or non-`private` methods as @Overrides can have strange and unforeseen consequences.

Let's make `mTokenEndpoint` `protected` and avoid this entirely

Reply:
Let's make `setTokenEndpoint` a `final` method -- should keep checkstyle happy while satisfying the previous comment